### PR TITLE
Revert #17547

### PR DESCRIPTION
**Pull request by roywei**
revert #17547 as it's causing performance regression.
See discussion in:
https://github.com/apache/incubator-mxnet/pull/17547#discussion_r413521892

### CODE REVIEW

**Comment by sxjscience** [comment]
@roywei @sandeep-krishnamurthy Is there a way to solve this issue? Should we try to accelerate the native dropout instead?

**Comment by roywei** [comment]
cc @ciyongch @TaoLv please don't include #17547  in 1.7.x branch as it will cause performance regression. I'm reverting in master for now, and probably won't have enough time to fix it before 1.7.x code freeze.

File: src/operator/nn/dropout-inl.h [hunk 1]
```diff
@@ -497,7 +492,7 @@ class DropoutOp {
   Context ctx_;
   cudnnDataType_t dtype_;
   cudnnDropoutDescriptor_t dropout_desc_;
-  uint64_t seed_;
+  uint64_t seed_ = 17 + rand() % 4096;  // NOLINT(runtime/threadsafe_fn)
```
Comment (leezu) [inline src/operator/nn/dropout-inl.h:495]: https://en.cppreference.com/w/cpp/numeric/random

**Comment by roywei** [comment]
Yes we should also consider implementing in cuda instead of cudnn, which will require major change on random module as well.
reference:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Dropout.cu

**Comment by ciyongch** [comment]
@roywei thanks for point it out, so far this PR is not included in either v1.x or v1.7.x.

**Pull request merged**